DQM/SiPixelPhase1RawData/src/SiPixelPhase1RawData.cc

```diff
@@ -0,0 +1,86 @@
+// -*- C++ -*-
+//
+// Package:     SiPixelPhase1RawData
+// Class:       SiPixelPhase1RawData
+//
+
+// Original Author: Marcel Schneider
+
+#include "DQM/SiPixelPhase1RawData/interface/SiPixelPhase1RawData.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "DataFormats/GeometryVector/interface/LocalPoint.h"
+
+
+SiPixelPhase1RawData::SiPixelPhase1RawData(const edm::ParameterSet& iConfig) :
+  SiPixelPhase1Base(iConfig) 
+{
+  srcToken_ = consumes<DetSetVector<SiPixelRawDataError>>(iConfig.getParameter<edm::InputTag>("src"));
+}
+
+
+void SiPixelPhase1RawData::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  edm::Handle<DetSetVector<SiPixelRawDataError>> input;
+  iEvent.getByToken(srcToken_, input);
+  if (!input.isValid()) return;
+
+  for (auto it = input->begin(); it != input->end(); ++it) {
+    for (auto& siPixelRawDataError : *it) { 
+      int fed = siPixelRawDataError.getFedId();
+      int type = siPixelRawDataError.getType();
+      DetId id = it->detId();
+      
+      const uint32_t LINK_bits = 6;
+      const uint32_t LINK_shift = 26;
+      const uint64_t LINK_mask = (1 << LINK_bits) - 1;
+
+      uint64_t errorWord = 0;
+      if (type == 32 || type == 33 || type == 34) {
```

Comment:
@davidlange6 I can't, I'm afraid. I just took the minmal subset needed to get the commonly used functionality out of https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DQM/SiPixelMonitorRawData/src/SiPixelRawDataErrorModule.cc . And that plugin is in a very sad shape, with lots more of these and a number of obvious mistakes (like the one fixed recently in 01341de52c2ee91e86e36).

For "real" phase1 we need to find some reliable documentation on the FED output and implement this properly.